pkg/solana/chainreader/chain_reader.go

```diff
@@ -187,17 +205,36 @@ func (s *SolanaChainReaderService) QueryKey(_ context.Context, _ types.BoundCont
 // Bind implements the types.ContractReader interface and allows new contract bindings to be added
 // to the service.
 func (s *SolanaChainReaderService) Bind(_ context.Context, bindings []types.BoundContract) error {
-	return s.bindings.Bind(bindings)
+	for _, binding := range bindings {
+		if err := s.bindings.Bind(binding); err != nil {
+			return err
+		}
+
+		s.lookup.bindAddressForContract(binding.Name, binding.Address)
+	}
+
+	return nil
```

Comment:
should Solana also have something like Binding registry to handle this in the same place>